### PR TITLE
Add is_type_compatible for Index & MultiIndex

### PR DESCRIPTION
**Pull request by itholic**
This PR proposes `is_type_compatible()` for `Index` & `MultiIndex`

```python
>>> kidx = ks.Index([1, 2, 3])
>>> kidx.is_type_compatible('integer')
True

>>> kidx = ks.Index([1.0, 2.0, 3.0])
>>> kidx.is_type_compatible('integer')
False
>>> kidx.is_type_compatible('floating')
True
```

### CODE REVIEW

File: databricks/koalas/indexes.py [hunk 1]
```diff
@@ -861,6 +861,24 @@ def is_object(self):
         """
         return is_object_dtype(self.dtype)
 
+    def is_type_compatible(self, kind):
```
Comment (HyukjinKwon) [inline databricks/koalas/indexes.py:864]: Can you add it to the docs?

Comment (itholic) [reply databricks/koalas/indexes.py:864]: pandas doesn't add this function to their docs.

Shall we add this to the docs regardless of it or keep excluding ?

Comment (HyukjinKwon) [reply databricks/koalas/indexes.py:864]: They have the doc page https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Index.is_type_compatible.html but seems it's not just listed up. Can you add it with hidden option as below?

```rst
.. toctree::
   :hidden:
   Index.is_type_compatible
```

Comment (itholic) [reply databricks/koalas/indexes.py:864]: Oh I didn't notice that hidden option. Thanks, I'll add it !! 👍

**Review by HyukjinKwon — APPROVED** [review]

**Pull request merged**